File: elan/jpa-study/src/main/java/kr/hoi9un/jpastudy/hw5/Member5.java
package kr.hoi9un.jpastudy.hw5;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.*;

@Entity
@Table(name = "members5")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Setter
@ToString
public class Member5 {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "team_id")
    private Team5 team;

    public Member5(String name) {
        this.name = name;
    }
}

File: joon.bug/src/main/kotlin/com/example/joonbug/chapter5/Member5.kt
package com.example.joonbug.chapter5

import jakarta.persistence.*

/**
 * 5장. 다대일 양방향 - Member 엔티티 (다 쪽, 연관관계 주인)
 */
@Entity
@Table(name = "chapter5_member")
class Member5(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var name: String = "",

    @ManyToOne
    @JoinColumn(name = "team_id")
    var team: Team? = null
) {
    /**
     * 무한루프 발생하는 toString (문제 2-1용)
     * Member.toStringWithLoop() -> Team.toStringWithLoop() -> Member.toStringWithLoop() -> ...
     */
    fun toStringWithLoop(): String {
        return "Member5(id=$id, name='$name', team=${team?.toStringWithLoop()})"
    }

    /**
     * 해결된 toString (문제 2-2용)
     * 연관 엔티티는 ID만 출력하거나 제외
     */
    override fun toString(): String {
        return "Member5(id=$id, name='$name', teamId=${team?.id})"
    }
}